src/ui/PickColor.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react'
+import { HexColorPicker } from 'react-colorful'
+import { setState, useStore } from '../store'
+
+export function PickColor(): JSX.Element {
+  const [color, setColor] = useState(useStore((state) => state.color))
+  const [pickcolor, set] = useStore((state) => [state.pickcolor, state.set])
+  const close = () => set((state) => ({ ...state, pickcolor: false }))
```

Comment:
```suggestion
  const close = setState({ pickcolor: false })
```

Reply:
hm...I tried the suggested change but didn't work..
`The expected type comes from property 'onClick' which is declared here on type 'DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>'`

This is how the ColorPicker is initialized
`<HexColorPicker color={color} onChange={setColor} />`

Reply:
Sorry, it should have been

```suggestion
  const close = () => setState({ pickcolor: false })
```

Reply:
Nice, that works. Thanks!
I need to learn a bit more, it's Zustand, right?

Reply:
yeah, it's zustand. It's pretty straightforward to use but there may still be some sub-optimal usage in the codebase. It's kind of easy to shoot yourself in the foot with.